src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,19 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    default String getPrintableString(Printable... printables) {
+        try {
+            StringBuilder builder = new StringBuilder();
+            for (Printable subPrintable : printables) {
+                builder.append(subPrintable.getPrintableString()).append(" ");
```

Comment:
Can try out doing this using StringJoiner which makes your code "prettier" 
Read about it [here](https://stackoverflow.com/questions/27522563/why-stringjoiner-when-we-already-have-stringbuilder)